src/main/java/com/algorand/algosdk/transaction/Transaction.java

```diff
@@ -951,14 +962,35 @@ public AssetParams(
 			   Address clawback) {
             if(assetTotal != null) this.assetTotal = assetTotal;
             this.assetDefaultFrozen = defaultFrozen;
-            if(assetUnitName != null) this.assetUnitName = assetUnitName;
-            if(assetName != null) this.assetName = assetName;
-            if(url != null) this.url = url;
-            if(metadataHash != null) this.metadataHash = metadataHash;
             if(manager != null) this.assetManager = manager;
             if(reserve != null) this.assetReserve = reserve;
             if(freeze != null) this.assetFreeze = freeze;
             if(clawback != null) this.assetClawback = clawback;
+
+            if(assetDecimals != null) {
+                if (assetDecimals < 0 || assetDecimals > 19) throw new RuntimeException("assetDecimals cannot be greater than 19");
```

Comment:
nit - add that they number should be also positive to the comment. Otherwise, if the user enters a negative number for some reason, s/he will get the wrong error.